src/main.cpp

```diff
@@ -2542,9 +2547,9 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
 
 bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationState& state, CBlockIndex * const pindexPrev)
 {
-    const Consensus::Params& consensusParams = Params().GetConsensus();
```

Comment:
Can you leave the consensusParams there? it will look redundant but it will get fixed after #5975 and #5968.
In fact maybe it makes sense for this to wait for #5975 and maybe use the ``` BlockMap& blockIndexMap``` param